rfc-102-add-notification-workflow.md

```diff
@@ -0,0 +1,26 @@
+# A 'notify' workflow step in publishing pipeline
+
+## Summary
+
+Email notifications are currently sent to relevant subscribers only when content is published with a major update.
+
+We've identified a use case where the significance of _organising_ content after it has been published may warrant an email notification.
+
+## Problem
+
+We have a high profile use case with the [business readiness finder](https://www.gov.uk/find-eu-exit-guidance-business) where the facet-tagging workflow step is deemed significant enough to warrant notifying subscribers via an email alert.
+
+Departments wish to notify users that new items of content from across GOV.UK have been included in the results for this finder. This is partly a consequence of tagging content after it has been published, but also in part due to the process of writing and reorganising guidance at pace.
+
+Republishing content for the purposes of notifying users is possible but this seems pollutive given the workflow event is (re)categorisation.
+
+The current (temporary) [tagging notification mechanism uses a direct HTTP request between Rummager and Email Alert API](https://github.com/alphagov/rummager/blob/master/lib/indexer/workers/metadata_tagger_notification_worker.rb#L11-L20).
+
+The current (temporary) [metadata tagging implementation is moving to Content Tagger](https://github.com/alphagov/content-tagger/pull/884), the publishing pipeline will handle [tagging via links](https://github.com/alphagov/govuk-content-schemas/commit/071731cb08e8c9c8956a70769748c979f762cf44).
+The current email tagging notifications will not be triggered once this has moved.
+
+## Proposal
+
+- Publishing API supports a 'notify' endpoint and sends a distinct message type downstream for Email Alert Service/API consumption.
```

Comment:
If I understand the user need, publishers want to be able to tag content, and have the subscribers to the finder be notified that the content of the finder has changed?

I guess one alternative to a separate, explicit "notify" endpoint would be to support something like this as a option when patching links.

I'm not sure that approach will work for finders though, I can see how it could work for Topic pages, as the content links specifically to the Topic page, but it doesn't sound like there's a direct link in the Publishing API sense between the content in the finder and the finder. Does that make sense?

Reply:
@cbaines it could well be integrated into another endpoint, we have one very awkward use case right now so keeping a clean separation seemed the pragmatic choice, if the need for this goes away I won't be sorry, by making a distinct thing it might be easier to deprecate and delete if it's ultimately superseded by better tagging workflow (ie as part of drafting and publishing) .

Reply:
Ok, sure :+1: